### ISSUE TITLE
Problem in COLMAP pipeline

### ISSUE THREAD

**Issue by stefan-baumann** [bennyguo/instant-nsr-pl#20]
I have been trying out some scenes with the COLMAP-based pipeline you provide in the repo and I've been running into the issue of the model learning only the input views and not being able to generalize at all.
I first suspected that this was an issue with my data, but even on the synthetic NeRF datasets that work very well with that respective pipeline (e.g. the lego bulldozer or the chair), I face this same issue.

This might look like this:
- Synthetic NeRF version (trained without the mask loss by setting the respective lambda to zero)
![image](https://user-images.githubusercontent.com/4400519/203110978-4543ec12-ef63-43cf-b449-923b6ec92bc0.png)
- COLMAP version (view from the training dataset - the views not in the train dataset show nothing useful)
![image](https://user-images.githubusercontent.com/4400519/203111014-78dd3c9e-4b5d-4d12-b97c-2c15ab449878.png)

One thing that I noticed is that the generated masks tend to be inverted compared to what one would expect - one extreme example for this is the drums scene (image from the training dataset again):
![image](https://user-images.githubusercontent.com/4400519/203111525-e28d3d55-5c6e-4b74-9b7a-343afef9111e.png)

Generally, I have experienced these issues with a wide variety of scenes (both from synthetic-NeRF & various real-world samples such as the dog used in the original NeuS publication) and with both your COLMAP preprocessing script & our internal one. I'm using the latest state of the repo with default configs except for the fused MLP being replaced with the vanilla one.

Have you been able to get some good samples out of that pipeline in your testing?
It would be awesome if you had some insights into this issue.

**Comment by bennyguo** [comment]
Sorry for the late reply. If just found that `model.radius` in `configs/neus-colmap.yaml` should be set to 0.5 as in `configs/nerf-colmap.yaml` (instead of 1.5). I've also noticed the floater problem and I'm experimenting several things to handle this.

**Comment by stefan-baumann** [comment]
Okay yeah, I guess that essentially ends up doing something very similar. The motivation for the factor of 4 (or rather $\sqrt{4^2 + 0.5^2}$) I mentioned was that this is the distance the cameras are placed at in the original dataset, so that should be a way to get equivalent behaviour in the end, but both ways work.

**Comment by bennyguo** [comment]
Please see #26 for my latest findings about improving quality and try our latest code :)

I think the floater problem is more likely to occur when training without masks, as extra blobs in background color do not harm the photometric loss. In SparseNeuS, an SDF penalty term is introduced to remove unused free surfaces:
<img width="917" alt="image" src="https://user-images.githubusercontent.com/19284678/205640666-09f555b6-fecc-4cba-88ad-d36657a22a0d.png">
You may try applying this penalty and see if things get improved.

**Comment by bennyguo** [comment]
See #26 for recent improvements about improving quality and removing floaters :)

**Comment by stefan-baumann** [comment]
Thank you very much! I will give it a try asap :)

**Comment by stefan-baumann** [comment]
I have a quick question about your implementation of the sparsity loss: you regularize the SDF values from the points where you already sample due to the main rendering. The original implementation uses both those points and randomly sampled points (see https://github.com/xxlong0/SparseNeuS/blob/50262f0fabfdd4f9ef695585dd431c34453e1098/models/trainer_generic.py#L808; in the finetuning script they go as far as further increasing the decay rate, see https://github.com/xxlong0/SparseNeuS/blob/62bd2670bc9f107d24e8546a3d5ecb62d32ef04d/models/trainer_finetune.py#L689). Did you find that this second term is not needed for good results or did you just exclude it for efficiency reasons?

**Comment by stefan-baumann** [comment]
I did not experiment with this in-depth yet, I can just say that the sdf decay as implemented in SparseNeuS did seem to help reduce floaters for at least one scene for me.

**Comment by bennyguo** [comment]
Great! I'll also experiment with sdf decay. Thanks for sharing this finding.
